internal/updater/updateable.go

```diff
@@ -46,13 +42,22 @@ func IsUpdateable(ctx context.Context) error {
 	if !fi.Mode().IsRegular() {
 		return fmt.Errorf("not a regular file")
 	}
-	if err := unix.Access(fn, unix.W_OK); err != nil {
-		return err
+	if err := canWrite(fn); err != nil {
+		return fmt.Errorf("can not write %q: %w", fn, err)
 	}
 
-	// check dir
-	fdir := filepath.Dir(fn)
-	return unix.Access(fdir, unix.W_OK)
+	// no need to check the directory since we'll be writing to the destination file directly
+	return nil
+}
+
+func canWrite(path string) error {
+	// check if we can write to this file. we open it for writing in
+	// append mode to not truncate it and immediately close the filehandle since
+	// we're only interested in the error.
+	fh, err := os.OpenFile(path, os.O_WRONLY|os.O_APPEND, 0755)
+	fh.Close()
```

Comment:
I would check the error on `Close`, just in case.

Reply:
Hmm, can `Close()` even fail when not writing anything to the file handle?